src/legacy/core_plugins/vis_type_table/public/agg_table/agg_table.js

```diff
@@ -134,12 +134,17 @@ export function KbnAggTable(config, RecursionHelper) {
                 formattedColumn.class = 'visualize-table-right';
               }
 
-              const isDate =
-                _.get(dimension, 'format.id') === 'date' ||
-                _.get(dimension, 'format.params.id') === 'date';
-              const isNumeric =
-                _.get(dimension, 'format.id') === 'number' ||
-                _.get(dimension, 'format.params.id') === 'number';
+              const checkDimensionType = (dimension, type) =>
+                _.get(dimension, 'format.id') === type ||
+                _.get(dimension, 'format.params.id') === type;
+
+              const shouldShowTotal = dimension => {
+                const allowedFormatTypes = ['number', 'duration', 'bytes', 'percent'];
+                return allowedFormatTypes.some(type => checkDimensionType(dimension, type));
+              };
+
+              const isDate = checkDimensionType(dimension, 'date');
+              const isNumeric = shouldShowTotal(dimension);
```

Comment:
This technique is pretty brittle- it says that there is a hardcoded list of formatters that can be used by the table, instead of looking at some property of the formatter. For example, I am adding new formatters: https://github.com/elastic/kibana/pull/53972

Is there a change we could make to the formatters API? We could take a different technique and add another property to number formats that would provide the formatted and raw data, but this might already exist?

Reply:
Hi @wylieconlon thank you for your comment, good point! How about adding additional property `isNumeric` (or maybe, `isSummable`) for all the types we need (`NumeralFormat` extending `NumberFormat`, `PercentFormat` and `BytesFormat` and, separately, `DurationFormat`? Please, re-check my changes and let me know what you think.